apps/journeys-admin/src/components/Editor/Slider/JourneyFlow/libs/arrangeSteps/arrangeSteps.ts

```
@@ -68,23 +88,127 @@ export function arrangeSteps(steps: TreeStepBlock[]): PositionMap {
   }
 
-  const hasActionBlocks = blocks
-    .flat()
-    .some((step) => filterActionBlocks(step).length > 0)
+  function buildBlocksByEdgeTraversal(stepEdges: Edge[]): void {
+    const sourceToTargets = buildSourceToTargetsMap(stepEdges)
+    const placedIds = new Set<string>()
+
+    function processLevel(levelSteps: TreeStepBlock[]): void {
+      blocks.push(levelSteps)
+      levelSteps.forEach((s) => placedIds.add(s.id))
```

Biome (2.3.14)

[error] This callback passed to forEach() iterable method should not return a value.

Either remove this return or remove the returned value.


(lint/suspicious/useIterableCallbackReturn)